### PR TITLE
`Development`: Add e2e playwright tests for exam announcement

### PR DESCRIPTION
**Pull request by muradium**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
We want to test exam announcement dialogs popping up by actions such as instructor sending a message or exam working time change.

### Description
Adds E2E tests in Playwright about exam announcements.

### Steps for Testing
- **Code Review**: Ensure that tests pass for valid reasons, functionality is adequately tested and check the code quality.
- **Run the tests (optional)**: Tests are checked by automatic run in CI environment. You can optionally run them locally and check if they pass.

**Steps for running the tests:**
1. Navigate to **src/tests/playwright**
2. Configure Playwright using playwright.env file based on your local setup. Current configuration should work for default Artemis setup.
3. Run `npm install && npm run playwright:setup && npx playwright test e2e/exam/ExamParticipation.spec.ts`
4. Confirm that **"Exam announcement"** tests pass

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/test/playwright/support/pageobjects/exam/ExamParticipationActions.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        await expect(this.page.locator('#exam-finished-title')).toContainText(title, { timeout: 40000 });
+    }
+
+    async checkExamFullnameInputExists() {
+        await expect(this.page.locator('#fullname')).toBeVisible({ timeout: 30000 });
+    }
+
+    async checkYourFullname(name: string) {
+        await expect(this.page.locator('#your-name')).toContainText(name, { timeout: 30000 });
+    }
```
Comment (rstief) [inline src/test/playwright/support/pageobjects/exam/ExamParticipationActions.ts:53]: Why are the timeouts so long? This would be 40/30 seconds, right?

**Review by rstief — APPROVED** [review]
Demonstrated in testing session, Code also LGTM

**Review by BBesrour — DISMISSED** [review]
Demonstrated in a testing session.

**Review by rstief — APPROVED** [review]

**Review by BBesrour — CHANGES_REQUESTED** [review]
re-approve

**Review by BBesrour — APPROVED** [review]
re-approve
(last one was a misclick :'))

**Review by coolchock — APPROVED** [review]
demonstrated in the testing session, looks good. I think you could also add another e2e test for problem statement updates in a follow-up PR.

File: src/test/playwright/support/pageobjects/exam/ExamParticipationActions.ts (hunk 1 above)
Comment (krusche) [reply src/test/playwright/support/pageobjects/exam/ExamParticipationActions.ts:53]: yes, please fix it in another PR

**Pull request merged**